api/src/services/tm_types/tm2.js

```diff
@@ -115,7 +115,7 @@ class TM2API {
         // not found
         return db('campaigns').insert(obj)
       } else {
-        return db('campaigns').where('tm_id', obj.tm_id).update(obj)
+        return db('campaigns').where('id', rows[0].id).update(obj)
```

Comment:
@kamicut Why is this necessary if the column is constrained to be unique?

Reply:
Because we might have multiple tasking managers, so we select the project depending on tasker_id _and_ tm_id

Reply:
@kamicut My comment here was in reference to `rows[0]`, since there should only be one item returned if the column has a uniqueness constraint, right?

Reply:
ah, because knex always returns an array even for a single row, so it's sure to be the first item in that array.